Source/WebCore/platform/cocoa/PublicSuffixStoreCocoa.mm

```diff
@@ -27,63 +27,64 @@
 #import "PublicSuffixStore.h"
 
 #import <pal/spi/cf/CFNetworkSPI.h>
+#import <wtf/URLHelpers.h>
 #import <wtf/cocoa/NSURLExtras.h>
+#import <wtf/text/StringHash.h>
 
 namespace WebCore {
 
-static bool isPublicSuffixCF(const String& domain)
+static bool isPublicSuffixCF(StringView domain)
 {
-    NSString *host = WTF::decodeHostName(domain);
-    return host && _CFHostIsDomainTopLevel((__bridge CFStringRef)host);
+    if (auto host = WTF::URLHelpers::mapHostName(domain.toStringWithoutCopying(), nullptr))
+        return !host->isNull() ? _CFHostIsDomainTopLevel(host->createCFString().get()) : _CFHostIsDomainTopLevel(domain.createCFStringWithoutCopying().get());
+    return false;
 }
 
 bool PublicSuffixStore::platformIsPublicSuffix(StringView domain) const
 {
-    auto domainString = domain.toStringWithoutCopying();
     {
         Locker locker { m_publicSuffixCacheLock };
-        if (m_publicSuffixCache && m_publicSuffixCache->contains(domainString))
+        if (m_publicSuffixCache && m_publicSuffixCache->contains<ASCIICaseInsensitiveStringViewHashTranslator>(domain))
```

Comment:
Based on the assumption that anyone who uses `HashSet<String, ASCIICaseInsensitiveHash>` would want this translation if they happened to pass in a StringView, I think we should probably take the time to build the infrastructure to make this automatic without needing to pass the translator explicitly.

One approach would be to extend the Hash argument to support exposing a set of "related" or "translatable" types and their associated translators. Then StringHash (and ASCIICaseInsensitiveHash, etc.) would then provide the means for `String`, `StringView`, `ASCIILiteral`, etc to all be used in the translatable functions (contains(), add(), etc.). This would probably also provide a way to replace the "smart pointer" specific overloads I added a long time ago, with a more generic replacement.

Very roughly, the has containers would get new functions that look something like:

```c++
    template<typename RelatedType> 
        requires IsRelatedType<HashArg , RelatedType>  
    iterator find(const RelatedType& key) const
    {
        return find<LookupTranslatorForRelatedType<HashArg, RelatedType>>(std::forward<RelatedType>(key)); 
    }
 ```

and StringHash, for example, would get something akin to:

```c++
    struct StringHash {
        ...
        using Supported = TypeList<
             Alt<String, StringTranslator>, 
             Alt<StringImpl*, StringImplTranslator>, 
             Alt<StringView, StringViewTranslator>, 
             Alt<ASCIILiteral, ASCIILiteralTranslator>
        >;
    };
```

(plenty of other ways to structure that, such as using overloading, but you get the idea).